tensorflow/lite/micro/tools/metrics/create_size_log_x86.sh

```diff
@@ -0,0 +1,139 @@
+ 
+# Utility function to build a target.
+# Parameters:
+# ${1}: binary target name such as keyworkd_benchmark
+# ${2}: build type such as RELEASE, DEFAULT
+# ${3}: target such as linux
+# ${4}: target architecture such as x86_64
+function build_target() {
+  local binary_target=$1
+  local build_type=$2
+  local target=$3
+  local target_arch=$4
+  readable_run make -j8 -f tensorflow/lite/micro/tools/make/Makefile build BUILD_TYPE=${build_type} TARGET=${target} TARGET_ARCH=${target_arch} ${binary_target}
+}
+
+# Utility function to profile a binary and report its size
+#Parameters:
+# ${1}: binary target path
+# ${2}: size log file name
+function profile_a_binary() {
+  local binary=${1}
+  local log=${2}
+
+  raw_size=$(size ${GEN_FILES_DIR}/${binary})
+  # Skip the title row
+  sizes=$(echo "${raw_size}" | sed -n '2 p')
+  text_size=$(echo "$sizes" | awk '{print $1}')
+  data_size=$(echo "$sizes" | awk '{print $2}')
+  bss_size=$(echo "$sizes" | awk '{print $3}')
+  total_size=$(echo "$sizes" | awk '{print $4}')
+
+  echo "${binary}, ${text_size}, ${data_size}, ${bss_size}, ${total_size}" >> ${log}
+}
+
+# Parameters:
+# ${1} - size log file name
+function start_size_report() {
+  local log=${1}  
+
+  echo "" >> ${log}
+  echo "${START_HEADER}"  >> ${log}
+  date >> ${log}
+  echo "Profiled at ${HEAD_SHA}" >> ${log}
+  echo "binary name, text, data, bss, total" >> ${log}
+
+  readable_run make -f tensorflow/lite/micro/tools/make/Makefile third_party_downloads
+}
+
+# Parameters:
+# ${1} - size log file name
+function end_size_report() {
+  local log=${1}  
+
+  echo "${END_HEADER}"  >> ${log}
+  echo "" >> ${log}
+}
+
+# Parameters:
+# ${1} - (size) log file name
+# ${2} - target binary name such as keyword_benchmark
+function report_size() {
+  local log=${1}
+  local binary=${2}
+
+  local target="linux"
+  local target_arch="x86_64"
+  local build_type="release"
+
+  build_target ${binary} ${build_type} ${target} ${target_arch}
+  local build_result=$?
+  local binary_path="${target}_${target_arch}_${build_type}/bin/${binary}"
+
+
+  if [[ ${build_result} != 0 ]]
+  then
+    # Here release build failed so mark failures and return appropriate error
+    # code.
+    echo "${binary} fail to build,">> ${log}
+    return ${build_result}
+  fi
+
+  # If build is successful, profile the size.
+  profile_a_binary ${binary_path} ${log}
+}
+
+###################################################
+### Start of main
+###################################################
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOT_DIR=${SCRIPT_DIR}/../../../../..
+GEN_FILES_DIR=${ROOT_DIR}/tensorflow/lite/micro/tools/make/gen/
+cd "${ROOT_DIR}"
+
+source tensorflow/lite/micro/tools/ci_build/helper_functions.sh
+
+START_HEADER="################## START ##############################"
+END_HEADER="#################### END ############################"
+HEAD_SHA=`git rev-parse HEAD`
+SIZE_LOG=${SCRIPT_DIR}/size_data/size_linux_x86_64_release.csv
```

Comment:
my guess is that we would benefit from a separate log file for each target. That will make it easier to write analysis scripts since they can be focused on a single target's log file instead of having to parse the combined log file and separate out per-binary logs.

Reply:
Done.